elastic/kibana
# [Advanced Settings] Integrate new Settings application into stateful Kibana

Closes https://github.com/elastic/kibana/issues/172922 

## Summary

This PR:
- Integrates the new Settings application (`packages/kbn-management/settings/application`) into stateful Kibana and removes the old `management_app` from the `src/plugins/advanced_settings` plugin.
- Adds support for section registry in the new Settings application, so that other plugins can add their own sections to the Advanced settings app.
- Adds functionality for disabling saving of settings based on the provided capabilities of the current user.

<img width="1352" alt="Screenshot 2024-01-23 at 16 46 03" src="https://github.com/elastic/kibana/assets/59341489/1f3b7088-58e2-46e8-a7dd-ae0fc346b4ba">

<br><br>

"Usage collection" section in Global settings:

<img width="1099" alt="Screenshot 2024-01-23 at 16 48 24" src="https://github.com/elastic/kibana/assets/59341489/ebc54ad5-348b-46dd-a047-b418ddc7ba4f">

### How to test

**Testing Advanced settings in stateful Kibana:**
1. Start Es with `yarn es snapshot` and Kibana with `yarn start`
2. Go to Stack Management -> Advanced Settings
3. Verify that the app functions correctly. Both tabs (for space and global settings) should be displayed, setting fields should be editable and saveable, etc.

**Testing the section registry**
Currently, `telemetry_management_section` is the only plugin that registers a section - the "Usage collection" section under the "Global settings" tab. This should work correctly in stateful Kibana.
1. Start Es with `yarn es snapshot --license=trial` and Kibana with `yarn start`
2. Go to Stack Management -> Advanced Settings and select the "Global settings" tab
3. Scroll down and verify that the "Usage collection" section is displayed and works as expected.

**Testing with different capabilities:**
1. Start Es with `yarn es snapshot` and Kibana with `yarn start`
2. Go to Stack Management -> Roles
3. Create a role that has "Read" access to Advanced settings and one that doesn't have any access.
4. Create users with each of these two roles.
5. Log in with these users and verify that the user with "Read" access can see the app but cannot edit it, and the user with no privileges cannot access the app.

**Testing Advanced settings in serverless Kibana:**
The Advanced settings app in serverless shouldn't be affected by these changes.
1. Start Es with `yarn es serverless` and Kibana with `yarn serverless-{es/oblt/security}`
2. Go to Management -> Advanced Settings
3. Verify that the app functions correctly. There shouldn't be any tabs as there are no spaces.
